### DISCUSSION TITLE
Question about touch time

### DISCUSSION THREAD

**Discussion by bupy7** [yiisoft/cache-file#40] [Q&A]
I have saw this line: https://github.com/yiisoft/cache-file/blob/master/src/FileCache.php#L163 Would you think this is safe way to store TTL? What about store one as first line in the file and use next lines as cached value like this:

```php
use SplFileObject;
use function serialize;
use function time;
use const LOCK_EX;
use const LOCK_UN;
use const PHP_EOL;

$file = new SplFileObject('/path/to/cache/file.bin', 'c+')

$file->flock(LOCK_EX);

$file->rewind();
$file->ftruncate(0);
$file->fwrite((string)(time() + 3600)); // TTL 1 hour
$file->fwrite(PHP_EOL);
$file->fwrite((string)serialize($value)); // cached value
$file->fflush();

$file->flock(LOCK_UN);
```

It's just a question because I had tried to find out whether can I use future time for time of creating of the file and didn't find any strong answer. But I think that time of creating should be time of creating and no more.

**Comment by vjik** [comment]
Documentation does not mention the restriction on the modification date. 

I think store date as modified time more performance solution than save and load date from file content.
